### PR TITLE
voomp

### PR DESCRIPTION
**Pull request by magcius**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -224,34 +267,166 @@ impl ArchiveIndex {
...
+            pub num_files: u32,
+            #[deku(endian = "little")]
+            pub content_flags: u32,
+            #[deku(endian = "little")]
+            pub locale_flags: u32,
+            #[deku(count = "num_files")]
+            pub file_id_delta_table: Vec<u32>,
+            #[deku(count = "num_files")]
+            pub file_entries: Vec<RootFileEntry>,
+        }
```
Comment (wgreenberg) [inline src/main.rs:337]: bro

File: src/main.rs [hunk 2]
```diff
@@ -224,34 +267,166 @@ impl ArchiveIndex {
...
+            pub hash_size_ckey: u8,
+            pub hash_size_ekey: u8,
+            pub page_size_ckey: u16,
+            pub page_size_ekey: u16,
+            pub page_count_ckey: u32,
+            pub page_count_ekey: u32,
+            #[deku(assert_eq = "0")]
+            _pad1: u8,
+            pub espec_page_size: u32,
+        }
```
Comment (wgreenberg) [inline src/main.rs:385]: i

File: src/main.rs [hunk 3]
```diff
@@ -224,34 +267,166 @@ impl ArchiveIndex {
...
+                #[derive(DekuRead, Debug)]
+                #[deku(endian = "big")]
+                struct Page {
+                    pub ekey_count: u8,
+                    #[deku(pad_bytes_before = "1")]
+                    pub size: u32, // Technically this is a 40-bit size value. We chop off the first byte here... hope it doesn't matter!
+                    pub ckey: EKey,
+                    #[deku(count = "ekey_count")]
+                    pub ekey: Vec<EKey>,
+                }
```
Comment (wgreenberg) [inline src/main.rs:409]: can i sanction a pull request

**Review by wgreenberg — APPROVED** [review]
looks fine except the parts that don't

File: src/main.rs (hunk 3 above)
Comment (magcius) [reply src/main.rs:409]: clippy didn't complain about this one fyi

**Pull request merged**